common/script/index.coffee

```diff
@@ -1382,14 +1391,16 @@ api.wrap = (user, main=true) ->
           user.stats.lvl++
           tnl = api.tnl(user.stats.lvl)
 
+          user.stats.hp = 50
+
+          return if user.stats.lvl > api.maxLevel
```

Comment:
(This is based on only reading the code, not testing it, so it might be wrong.) 
I think this `return ...` should be `if user.stats.lvl <= api.maxLevel` wrapped around the allocation code, so that allocation won't occur above level 100 but the rest of the subroutine will keep on happening. We want to keep looping through the `while` loop until we've done as much levelling up as needed, and we want to execute the `user.stats.exp = stats.exp` line after the while loop is done.